src/actions/BitPayActions.js

```diff
@@ -117,30 +147,38 @@ export async function launchBitPay(dispatch: Dispatch, uri: string, currencyWall
   const instructionOutput = invoiceInstruction.outputs[0]
 
   // Make the spend to generate the tx hexes
+  const requiredFeeRate = invoiceInstruction.requiredFeeRate
   const spendInfo: EdgeSpendInfo = {
     currencyCode,
-    customNetworkFee: { satsPerByte: invoiceInstruction.requiredFeeRate },
     spendTargets: [
       {
         nativeAmount: instructionOutput.amount.toString(),
         publicAddress: instructionOutput.address
       }
-    ]
+    ],
+    otherParams: {
+      paymentProtocolInfo: {
+        merchant: {
+          requiredFeeRate
+        }
+      }
+    }
   }
-  const wallet = currencyWallets[walletId]
-  const unsignedTx = await wallet.makeSpend(spendInfo)
+  // $FlowFixMe
+  const unsignedTx = await selectedWallet?.makeSpend(spendInfo)
   const unsignedHex = unsignedTx.otherParams?.txJson?.hex ?? ''
-  const signedTx = await wallet.signTx(unsignedTx)
+  // $FlowFixMe
+  const signedTx = await selectedWallet?.signTx(unsignedTx)
   const signedHex = signedTx.signedTx ?? ''
 
-  errorData = { ...errorData, spendInfo, walletId, unsignedTx, signedTx }
+  errorData = { ...errorData, spendInfo, selectedWallet, unsignedTx, signedTx }
```

Comment:
An EdgeCurrencyWallet has keys so it's a bad idea to attach it to an object that may be logged. Passing the walletId is a better option.